pkg/controlplane/controller.go

```diff
@@ -236,141 +163,3 @@ func (c *Controller) RunKubernetesNamespaces(ch chan struct{}) {
 		}
 	}, c.SystemNamespacesInterval, ch)
 }
-
-// RunKubernetesService periodically updates the kubernetes service
-func (c *Controller) RunKubernetesService(ch chan struct{}) {
-	// wait until process is ready
-	wait.PollImmediateUntil(100*time.Millisecond, func() (bool, error) {
-		var code int
-		c.client.CoreV1().RESTClient().Get().AbsPath("/readyz").Do(context.TODO()).StatusCode(&code)
-		return code == http.StatusOK, nil
-	}, ch)
-
-	wait.NonSlidingUntil(func() {
-		// Service definition is not reconciled after first
-		// run, ports and type will be corrected only during
-		// start.
-		if err := c.UpdateKubernetesService(false); err != nil {
-			runtime.HandleError(fmt.Errorf("unable to sync kubernetes service: %v", err))
-		}
-	}, c.EndpointInterval, ch)
-}
-
-// UpdateKubernetesService attempts to update the default Kube service.
-func (c *Controller) UpdateKubernetesService(reconcile bool) error {
-	// Update service & endpoint records.
-	// TODO: when it becomes possible to change this stuff,
-	// stop polling and start watching.
-	// TODO: add endpoints of all replicas, not just the elected master.
```

Comment:
this TODO are addressed with this PR, the endpoints TODO is already handled by the endpoints reconciler